docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
+
+Within `RTNCalculator`, create three subfolders: `js`, `ios`, and `android`.
+
+The final result should look like this:
+
+```sh
+TurboModulesGuide
+├── MyApp
+└── RTNCalculator
+    ├── android
+    ├── ios
+    └── js
+```
+
+## 2. JavaScript Specification
+
+The **New Architecture** requires interfaces specified in a typed dialect of JavaScript (either [Flow](https://flow.org/) or [TypeScript](https://www.typescriptlang.org/)). **Codegen** will use these specifications to generate code in strongly-typed languages, including C++, Objective-C++, and Java.
+
+There are two requirements the file containing this specification must meet:
+
+1. The file **must** be named `Native<MODULE_NAME>`, with a `.js` or `.jsx` extension when using Flow, or a `.ts`, or `.tsx` extension when using TypeScript. Codegen will only look for files matching this pattern.
+2. The file must export a `TurboModuleRegistrySpec` object.
+
+<Tabs groupId="turbomodule-specs" defaultValue={constants.defaultJavaScriptSpecLanguages} values={constants.javaScriptSpecLanguages}>
+<TabItem value="flow">
+
+```typescript title="NativeCalculator.js"
+// @flow
+import type { TurboModule } from 'react-native/Libraries/TurboModule/RCTExport';
+import { TurboModuleRegistry } from 'react-native';
+
+export interface Spec extends TurboModule {
+  add(a: number, b: number): Promise<number>;
+}
+export default (TurboModuleRegistry.get<Spec>(
+  'RTNCalculator'
+): ?Spec);
+```
+
+</TabItem>
+<TabItem value="typescript">
+
+```typescript title="NativeCalculator.ts"
+import type { TurboModule } from 'react-native/Libraries/TurboModule/RCTExport';
+import { TurboModuleRegistry } from 'react-native';
+
+export interface Spec extends TurboModule {
+  add(a: number, b: number): Promise<number>;
+}
+
+export default (TurboModuleRegistry.get<Spec>(
+  'RTNCalculator'
+) as Spec | null);
+```
+
+</TabItem>
+</Tabs>
+
+At the beginning of the spec files are the imports:
+
+- The `TurboModule` type, which defines the base interface for all TurboModules
+- The `TurboModuleRegistry` JavaScript module, which contains functions for loading TurboModules
+
+The second section of the file contains the interface specification for the TurboModule. In this case, the interface defines the `add` function which takes two numbers and returns a promise that resolves to a number. This interface type **must** be named `Spec` for a TurboModule.
+
+Finally, we invoke `TurboModuleRegistry.get`, passing the module's name, which will load the TurboModule if it's available. 
+
+:::caution
+We are writing JavaScript files importing types from libraries, without setting up a proper node module and installing its dependencies. Your IDE will not be able to resolve the import statements and you may see errors and warnings. This is expected and will not cause problems when you add the module to your app.
+:::
```

Comment:
Nit: should we move the `package.json` above so this warning is not necessary. Then in the `Module Configuration` section we just focus on configuring the `codegenConfig`?

Reply:
> should we move the package.json above so this warning is not necessary.

that doesn't stop VSCode from generating warnings, for me.